portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java

```diff
@@ -89,6 +89,11 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
         // if we want to add comment, then there's no commentID
         if ("add".equals(action)) {
             String text = request.getParameter(Comments.TEXT);
+            if (text == null) {
+                response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Parameter 'text' doesn't exist, required for action 'add'");
+                return;
+            }
```

Comment:
nit: instead of having the early return, it might be better to simply use an if/else here...

```
if (text == null) {
                response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Parameter 'text' doesn't exist, required for action 'add'");
                return;
            }
Commenting on lines +92 to +95
 

 
nit: instead of having the early return, it might be better to simply use an if/else here...

```
if (text == null) {
  response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Parameter 'text' doesn't exist, required for action 'add'");
} else {
  comments.addComment(text);
  response.sendRedirect(commentsPage);
}
return;